### PR TITLE
Issue: Leftmost Repeating Character

### PR DESCRIPTION
**Pull request by AastikSharma05**
<hr>

## Description 📜
Given a string, find the repeated character present first in the string.
Fixes #2959 

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Competitive Programming

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

**Comment by AastikSharma05** [comment]
Please review my PR and let me know if any change is required.
Thanks for assigning me this issue :)

File: Competitive_Programming/C++/String/Leftmost_Repeating_Character.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
+Repeated Character Whose First Appearance is Leftmost
+
+Given a string, find the repeated character present first in the string.
```
Comment (ravireddy07) [inline Competitive_Programming/C++/String/Leftmost_Repeating_Character.md:3]: Add a title as problem Statement

File: Competitive_Programming/C++/String/Leftmost_Repeating_Character.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
+Repeated Character Whose First Appearance is Leftmost
```
Comment (ravireddy07) [inline Competitive_Programming/C++/String/Leftmost_Repeating_Character.md:1]: Add a title

File: Competitive_Programming/C++/String/Leftmost_Repeating_Character.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+Given a string, find the repeated character present first in the string.
+
+Examples:  
+
+Input  : geeksforgeeks
+Output : g
+(mind that it will be g, not e.)
```
Comment (ravireddy07) [inline Competitive_Programming/C++/String/Leftmost_Repeating_Character.md:9]: You can point it out in more better way

File: Competitive_Programming/C++/String/Leftmost_Repeating_Character.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	int index = firstRepeating(str);
+	if (index == -1)
+		printf("Either all characters are "
+			"distinct or string is empty");
+	else
+		printf("First Repeating character"
+			" is %c",
+			str[index]);
+	return 0;
+}
```
Comment (ravireddy07) [inline Competitive_Programming/C++/String/Leftmost_Repeating_Character.md:69]: seems like not everything is inserted into code. Always preview the `.md`.

File: Competitive_Programming/C++/String/Leftmost_Repeating_Character.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+### Output
+Output: 
+First Repeating character is g
```
Comment (ravireddy07) [inline Competitive_Programming/C++/String/Leftmost_Repeating_Character.md:72]: highlight this

File: Competitive_Programming/C++/String/Leftmost_Repeating_Character.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+ 
+### Complexity analysis
+Time Complexity : O(n). It does only one traversal of input string.
```
Comment (ravireddy07) [inline Competitive_Programming/C++/String/Leftmost_Repeating_Character.md:75]: `It does only one traversal of input string.`
Add this sentance as a commit.

**Review by ravireddy07 — CHANGES_REQUESTED** [review]
Please make suggested changes and make sure you preview the file and verify it from your side. @AastikSharma05

**Comment by AastikSharma05** [comment]
@ravireddy07 I have made the changes please check if anything else needs to be done.

File: Competitive_Programming/C++/String/Leftmost_Repeating_Character.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
+### Repeated Character Whose First Appearance is Leftmost (Strings)
```
Comment (ravireddy07) [inline Competitive_Programming/C++/String/Leftmost_Repeating_Character.md:1]: Keep title with `##` or `#`

**Review by ravireddy07 — CHANGES_REQUESTED** [review]
Refer this below images of you doc:

Image1: The header file is out of scope, use correct syntax of how to insert code in markdown.

image1

Image2: don't put everyting in one line.

image2

**Comment by ravireddy07** [comment]
@AastikSharma05  Please preview the `markdown` before you make a push.

**Comment by AastikSharma05** [comment]
@ravireddy07 I have modified as advised by you. Please check if it is as per your expectations.

File: Competitive_Programming/C++/String/Leftmost_Repeating_Character.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+```
+Input: geeksforgeeks
+Output: 
+First Repeating character is g
+ ```
+### Complexity analysis
+
+Time Complexity : O(n).
+
+It does only one traversal of input string.
```
Comment (ravireddy07) [inline Competitive_Programming/C++/String/Leftmost_Repeating_Character.md:159]: Add this line as comment, as it refering.

like:

```
Time Complexity : O(n).   #It does only one traversal of input string.
Auxiliary Space : O(1)
```

File: Competitive_Programming/C++/String/Leftmost_Repeating_Character.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+Input: geeksforgeeks
+Output: 
+First Repeating character is g
+``` 
+ 
+### Complexity analysis
+
+Time Complexity : O(n).
+
+It does only one traversal of input string.
```
Comment (ravireddy07) [inline Competitive_Programming/C++/String/Leftmost_Repeating_Character.md:92]: here as well

**Review by ravireddy07 — COMMENTED** [review]
Great work @AastikSharma05 ,
update this tiny change and we're good to merge.

**Review by ravireddy07 — APPROVED** [review]

**Pull request merged**